### DISCUSSION TITLE
sensors_launch.py on docker keeps resulting in *** buffer overflow detected***

### DISCUSSION THREAD

**Discussion by bdawkinsroj** [iRobotEducation/create3_docs#600] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

latest humble release

### Which RMW is your robot running?

I don't know

### Does your robot have an assigned namespace? If so, please share.

no

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

Yesterday I got my slam toolbox fixed, but now the sensors launch is causing problems. I couldn't find a solution that worked anywhere. I am running this on docker on my pi 5 and I am able to use the lidar perfectly fine when I test it on my pc. The output for the error is:

`[rplidar_composition-2] *** buffer overflow detected ***: terminated
[ERROR] [rplidar_composition-2]: process has died [pid 2296, exit code -6, cmd '/opt/ros/humble/lib/rplidar_ros/rplidar_composition --ros-args -r __ns:=/ --params-file /create3_ws/install/create3_lidar_slam/share/create3_lidar_slam/config/rplidar_node.yaml'].`

**Comment by alsora** [comment]
This seems to be a problem in how the lidar is connected, see: https://github.com/Slamtec/rplidar_ros/issues/63
You could try the following (or whatever the device where the lidar appears is in your case).
```
sudo chmod 777 /dev/ttyUSB0
```
Note that the presence of docker may also make it more complex; You may need to allow docker access to the connected devices.

If that doesn't help, could you run the `rplidar` node outside of roslaunch with `gdb` and share a backtrace?
For example:

```
gdb --args /opt/ros/humble/lib/rplidar_ros/rplidar_composition --ros-args -r __ns:=/ --params-file /create3_ws/install/create3_lidar_slam/share/create3_lidar_slam/config/rplidar_node.yaml
```
You may need to install `gdb` if it's not already in your docker.

**Reply by bdawkinsroj** [reply]
chmod hasn't worked for me, however I was able to gain access to the USB with:
`docker run -t -i --device=/dev/ttyUSB0 ubuntu bash`
but it wouldn't let me make the container with my image so it seems useless. Do you know how to build the container with these settings and the image?
